keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/SecurityGroupHandler.kt

```diff
@@ -306,6 +308,14 @@ class SecurityGroupHandler(
         .toSet()
     )
 
+  private fun SecurityGroupModel.SecurityGroupRulePortRange.toPortRange(): IngressPorts =
+    (startPort to endPort).let { (start, end) ->
+      if (start == null || end == null) AllPorts else PortRange(start, end)
```

Comment:
I'm impressed that the Kotlin compiler is smart enough to know that the arguments to `PortRange` are not nullable.

Reply:
That's why I had to use that funky `let` construction. It wouldn't do it directly with the properties of the `SecurityGroupRulePortRange` class because it can't tell if there's anything side-effecty going on, or another thread messing with the object concurrently.

Reply:
Not just for null handling either: https://kotlinlang.org/docs/reference/typecasts.html#smart-casts